chronos/internal/asyncfutures.nim

```diff
@@ -1264,7 +1297,7 @@ proc sleepAsync*(duration: Duration): Future[void] =
   return retFuture
 
 proc sleepAsync*(ms: int): Future[void] {.
-     inline, deprecated: "Use sleepAsync(Duration)".} =
+     inline, deprecated: "Use sleepAsync(Duration)", asyncraises: [CancelledError].} =
   result = sleepAsync(ms.milliseconds())
 
 proc stepsAsync*(number: int): Future[void] =
```

Comment:
Why you missed this one? It still could raise `CancelledError` too, actually every async procedure could raise it, either raw or transformed one could raise `CancelledError`.

Reply:
I wasn't sure if it can raise but I guess it can.

Raw or not, if an `async` function always returns a completed Future, it doesn't ever raise `CancelledError`.

This effectively means that only functions (raw or not) that create a future manually and have a non-finishing code path can ever raise `CancelledError` (plus those that add `CancelledError` manually for future-proofing the API).

Reply:
Also, notably, any function that doesn't have an `asyncraises` is implicitly treated as `asyncraises: [CatchableError]`